homeassistant/components/knx/light.py

```diff
@@ -392,3 +519,53 @@ async def set_color(
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the light off."""
         await self._device.set_off()
+
+
+class KnxYamlLight(_KnxLight):
+    """Representation of a KNX light."""
+
+    _device: XknxLight
+
+    def __init__(self, xknx: XKNX, config: ConfigType) -> None:
+        """Initialize of KNX light."""
+        super().__init__(_create_yaml_light(xknx, config))
+        self._attr_max_color_temp_kelvin: int = config[LightSchema.CONF_MAX_KELVIN]
+        self._attr_min_color_temp_kelvin: int = config[LightSchema.CONF_MIN_KELVIN]
+        self._attr_entity_category = config.get(CONF_ENTITY_CATEGORY)
+        self._attr_unique_id = self._device_unique_id()
+
+    def _device_unique_id(self) -> str:
+        """Return unique id for this device."""
+        if self._device.switch.group_address is not None:
+            return f"{self._device.switch.group_address}"
+        return (
+            f"{self._device.red.brightness.group_address}_"
+            f"{self._device.green.brightness.group_address}_"
+            f"{self._device.blue.brightness.group_address}"
+        )
+
+
+class KnxUiLight(_KnxLight):
+    """Representation of a KNX light."""
+
+    _device: XknxLight
+    _attr_has_entity_name = True
+
+    def __init__(
+        self, knx_module: KNXModule, unique_id: str, config: ConfigType
+    ) -> None:
+        """Initialize of KNX light."""
+        super().__init__(
+            _create_ui_light(
+                knx_module.xknx, config[DOMAIN], config[CONF_ENTITY][CONF_NAME]
+            )
+        )
+        self._attr_max_color_temp_kelvin: int = config[DOMAIN][CONF_COLOR_TEMP_MAX]
+        self._attr_min_color_temp_kelvin: int = config[DOMAIN][CONF_COLOR_TEMP_MIN]
+
+        self._attr_entity_category = config[CONF_ENTITY][CONF_ENTITY_CATEGORY]
+        self._attr_unique_id = unique_id
+        if device_info := config[CONF_ENTITY].get(CONF_DEVICE_INFO):
+            self._attr_device_info = DeviceInfo(identifiers={(DOMAIN, device_info)})
+
+        knx_module.config_store.entities[unique_id] = self
```

Comment:
Don't store the entity in a shared container. The entity should never be accessed from outside its platform.

Reply:
👍
One of the usecases was updating entity configuration. `Entity.async_remove()` was called with this reference, before `async_add_entity()` reinstantiates it. How would this have to be done without a reference. I think using `entity_registry.async_remove(entity_id)` would mark it as deleted in the registry. Is there another way to call this from eg. the entity registry that I'm missing?
Or is it safe to store a reference / partial to `Entity.async_remove()` and call this, instead of the `Entity` itself?

Reply:
We shouldn't call the Entity API directly from outside. If we don't want to remove the entity entry from the entity registry we can use the dispatch helper to signal the removal of the Entity instance.

Reply:
addressed in #123038 at 1163e34586bfd388a6a95f197a1af2d7a38f78ad